src/lib/y2storage/proposal/space_maker.rb

```diff
@@ -173,24 +193,63 @@ def success?(planned_partitions)
       # @param lvm_helper [Proposal::LvmHelper] contains information about how
       #     to deal with the existing LVM volume groups
       # @param disk [String] optional disk name to restrict operations to
+      #
+      # rubocop:disable Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+      #
       def resize_and_delete!(planned_partitions, keep, lvm_helper, disk: nil)
-        log.info "Resize and delete. disk: #{disk}"
+        log.info "Resize and delete. disk: #{disk}, planned partitions:"
+        planned_partitions.each do |p|
+          log.info "  mount: #{p.mount_point}, disk: #{p.disk}, min: #{p.min}, max: #{p.max}"
+        end
 
+        # restart evaluation
         @distribution = nil
 
-        # Initially, resize only if there are no Linux partitions in the disk
+        # maybe it works already...
+        return if success?(planned_partitions)
+
+        # Try various methods to free space. Stop when a valid partition
+        # layout has been found.
```

Comment:
If we slightly change the arguments order for `delete_partitions!` and `delete_disk_content`, we can write this part like this. That should please both rubocop and the human readers (as a human, I even prefer it).

```ruby
# Try various methods to free space. Stop when a valid partition
# layout has been found.
methods_and_args = [
  # step 1 - resize Windows partitions only if there are no Linux partitions on the disk
  [:resize_windows!,     [{ force: false }]],
  # step 2 - # delete Linux partitions
  [:delete_partitions!,  [:linux, keep]], 
  # step 3 - delete other (non-Windows) partitions
  [:delete_partitions!,  [:other, keep]], 
  # step 4 - resize Windows partitions
  [:resize_windows!,     [{ force: true }]],
  # step 5 - delete Windows partitions
  [:delete_partitions!,  [:windows, keep]], 
  # step 6 - if deleting partitions was not enough, maybe there is no
  # partition table and we have to wipe the disk
  [:delete_disk_content, [lvm_helper]]
]

methods_and_args.each_with_index do |method_and_args, idx|
  method, extra_args = method_and_args

  log.info "Step #{idx + 1} - #{method}"
  send(method, planned_partitions, disk, *extra_args)

  break if @distribution
end

raise Error unless @distribution
```